apps/meteor/server/lib/getClientAddress.ts

```diff
@@ -1,8 +1,12 @@
-import type { IMethodConnection } from '@rocket.chat/core-typings';
+import type { ISocketConnection } from '@rocket.chat/core-typings';
 
-export function getClientAddress(connection: IMethodConnection): string {
+export const getClientAddress = (connection: Pick<ISocketConnection, 'clientAddress' | 'httpHeaders'>): string => {
 	if (!connection) {
 		return '';
 	}
-	return connection.clientAddress || connection.httpHeaders['x-real-ip'];
-}
+
+	const { clientAddress, httpHeaders } = connection;
+	const xRealIp = Array.isArray(httpHeaders?.['x-real-ip']) ? httpHeaders['x-real-ip'][0] : httpHeaders['x-real-ip'];
+
+	return clientAddress || xRealIp || '';
+};
```

Comment:
Why this change? x-real-ip is never an array, did you encounter it being so?

Reply:
When we use **ISocketConnection** the object **httpHeaders** can be `string | string[] | undefined`